### ISSUE TITLE
Use SEEK_DATA/SEEK_HOLE to skip reading holes

### ISSUE THREAD

**Issue by crass** [doug-gilbert/ddpt#6]
For files that have large holes in them, it makes little sense to read those holes knowing they will be zero-filled. For instance, when using `sparse` mode, ddpt could seek to the end of a hole in the input with SEEK_DATA. Of course, the user could figure out the data blocks themself and use a scatter-gather list to only read the data. But it would be nice to have it be transparent.

It looks like similar functionality could be implemented for SCSI devices via the `SCSI GET LBA STATUS` command to see what blocks are mapped.

**Comment by doug-gilbert** [comment]
Yes, I planning to do this. My plan is to add a new executable to the ddpt package, called ddpt_sparse which by default will have sparse reading (on input) and sparse writing (on output). Currently ddpt only does sparse writing if oflag=sparse is given. ddpt_sparse will be an extension of ddpt_sgl as it already exists and is written in C++.
